File: learntoprogram/ExtendsDemo.java
/**
* Description:
* 继承
* @author fnfly2005
* @version 1.0
*/
class Persion
{
    int age = 0;
    final String COUNTRY = "CN";//final 关键词修饰的变量、方法、类具有只读属性，不可被修改、继承、覆盖

    String name = "fnfly2005";

    Persion()
    {
        System.out.println("哇哇");
    }

    void show()
    {
        System.out.println(this.age+"..fu.."+this.age);
    }

}

class Student extends Persion
{
    int age = 10;

    Student()
    {
        //省略父类的空参数构造函数super(),父类构造函数不可被覆盖
    }

    /**
        this代表一个本类对象的引用，super代表父类空间
        覆盖父类方法,子类权限大于等于父类,静态只能覆盖静态
    */
    void show()
    {
        super.show();//保留原有功能
        System.out.println(this.age+"..zi.."+super.age);//新增扩展功能
    }

    void study()
    {
        System.out.println(name+"....student study");
    }
}

class Worker extends Persion
{
    void work()
    {
        System.out.println(name+"....worker work");
    }
}

class ExtendsDemo
{
    public static void main(String[] args)
    {
        Student s = new Student();
        Worker w = new Worker();
        s.show();
        w.show();
        s.study();
    }
}

File: learntoprogram/ExtendsDemo.py
#!/usr/bin/python
#coding:utf-8
'''
Description:继承
Version: v1.0
'''
class Persion():

    def __init__(self):
        self.age = 20
        self.name = 'fnfly2005'
        print "哇哇"

    def show(self):
        print str(self.age) + "..fu.." + str(self.age)

class Student(Persion):
    #覆盖父类构造函数
    def __init__(self):
        Persion.__init__(self)#调用父类构造函数
        self.age = 10#会覆盖父类同名属性

    def show(self):
        Persion.show(self)#调用父类方法
        print str(self.age) + "..zi.." + str(self.age)#新增扩展功能

    def study(self):
        print self.name + "....student study"#调用父类属性,已在构造函数中初始化

class Worker(Persion):
    #子类没有构造函数，将继承父类的构造函数
    def work(self):
       print self.name +"....worker work"

if __name__ == '__main__':
    s=Student()
    s.show()
    s.study()
    w=Worker()
    w.show()
    w.work()
